docs/en/how_to_guides/finetune_with_a_custom_dataset.md

```diff
@@ -0,0 +1,357 @@
+    # read the annotation file to get the label of each image
+    def annotations(annotation_file_path):
+        image_label = {}
+        for i in open(annotation_file_path, "r"):
+            label = " ".join(i.split(" ")[1:]).replace("\n", "").replace("/", "_")
+            if label not in image_label.keys():
+                image_label[label] = []
+                image_label[label].append(i.split(" ")[0])
+            else:
+                image_label[label].append(i.split(" ")[0])
+        return image_label
+
+    # make a new folder for subset
+    subset_path = images_path + subset_name
+    os.mkdir(subset_path)
+
+    # extract and copy images to the new folder above
+    image_label = annotations(annotation_file_path)
+    for label in image_label.keys():
+        label_folder = subset_path + "/" + label
+        os.mkdir(label_folder)
+        for image in image_label[label]:
+            image_name = image + ".jpg"
+            shutil.copy(images_path + image_name, label_folder + image_name)
+
+
+images_path = "./aircraft/data/images/"
+subset_name = "trainval"
+annotation_file_path = "./aircraft/data/images_variant_trainval.txt"
+extract_images(images_path, subset_name, annotation_file_path)
+```
+
+The splitting method of the test set is the same as that of the training set. The file structure of the whole Aircraft dataset should be:
+
+```text
+aircraft
+└── data
+	└── images
+        ├── trainval
+        │   ├── 707-320
+        │   │   ├── 0056978.jpg
+        │   │   └── ....
+        │   ├── 727-200
+        │   │   ├── 0048341.jpg
+        │   │	└── ....
+        │  	└── ....
+        └── test
+            ├── 707-320
+            │   ├── 0062765.jpg
+            │   └── ....
+            ├── 727-200
+            │   ├── 0061581.jpg
+            │   └── ....
+            └── ....
+```
+
+./example/finetune.py integrates the whole training pipeline, from pre-processing to the establishment and training of the model: `create_dataset` -> `create_transforms` -> `create_loader` -> `create_model` ->..., thus, the dataset with the adequate file directory structure can be sent directly to the fine-tuning script to start the subsequent processes including loading dataset and model training by running `python ./example/finetune.py --data_dir=./aircraft/data/images/` command. For custom datasets, please note that<font color=DarkRed> the dataset parameter in the configuration file must be set to an empty string `""` </font> in advance.
+
+#### Read Dataset online
+
+Offline data reading takes up extra local disk space to store the newly generated data files. Therefore, when the local storage space is insufficient or the data cannot be backed up to the local environment, the local data files cannot be read using `create_dataset` directly, you can write a function to read the dataset in an online way.
+
+Here's how we generate an iterable object that stores the images of the training set:
```

Comment:
I think the source dataset defined below is a mappable dataset instead of an iterable dataset( which should have __iter__ method.

Reply:
How about "an random-accessible dataset object that stores the images of the training set and realize a map from indices to data samples"? the "random-accessible dataset" term is from mindspore site.